Master analyze_base.dart to null-safety

WizzXu opened the pull request:
migrate analyze_base.dart to null-safety
List which issues are fixed by this PR. You must list at least one issue.
#71511

If you had to change anything in the flutter/tests repo, include a link to the migration guide as per the breaking change policy.
NO

Pre-launch Checklist

- [x]  I read the Contributor Guide and followed the process outlined there for submitting PRs.
- [x]  I read the Tree Hygiene wiki page, which explains my responsibilities.
- [x]  I read and followed the Flutter Style Guide, including Features we expect every widget to implement.
- [x]  I signed the CLA.
- [x]  I listed at least one issue that this PR fixes in the description above.
- [x]  I updated/added relevant documentation (doc comments with ///).
- [x]  I added new tests to check the change I am making, or this PR is test-exempt.
- [x]  All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on Discord.

goderbauer committed:
Leader not always dirty (#92598)

engine-flutter-autoroll committed:
Roll Plugins from 93f04832c622 to 34ea0c3c1a09 (1 revision) (#92677)

engine-flutter-autoroll committed:
Roll Engine from c9dda84396f2 to 191b18d49026 (3 revisions) (#92680)

engine-flutter-autoroll committed:
Roll Engine from 191b18d49026 to 9295037358f5 (1 revision) (#92684)

engine-flutter-autoroll committed:
Roll Engine from 9295037358f5 to 37f1b478eed3 (1 revision) (#92688)

engine-flutter-autoroll committed:
Roll Engine from 37f1b478eed3 to a04ec3aaa6f1 (3 revisions) (#92691)

Hixie committed:
Feature/cpu gpu memory gallery transition tests (#92626)

engine-flutter-autoroll committed:
Roll Plugins from 34ea0c3c1a09 to 03622197cc1e (1 revision) (#92690)

engine-flutter-autoroll committed:
Roll Engine from a04ec3aaa6f1 to ddf4bd598eb3 (3 revisions) (#92694)

zhongwuzw committed:
[flutter_tools] [iOS] Change UIViewControllerBasedStatusBarAppearance to true to fix rotation status bar disappear in portrait (#92647)

HansMuller committed:
InputDecorator floating label origin no longer depends on ThemeData.fixTextFieldOutlineLabel (#92698)

guidezpl committed:
Refactor ThemeData  (#91497)

christopherfujino committed:
wrap $DART_BIN invocation in double quotes to escape spaces (#92617)

sunyujia-rakuten committed:
Fix typo in the API docs of brightness (#92713)

wanglikun7342 committed:
feat: migrate test_data/project.dart to null safety (#92646)

Hixie committed:
Improve how AttributedStrings are presented in the widget inspector (#92450)

engine-flutter-autoroll committed:
Roll Plugins from 03622197cc1e to e51cc1df7b75 (1 revision) (#92724)

engine-flutter-autoroll committed:
Roll Engine from ddf4bd598eb3 to 3a0a63c2d209 (12 revisions) (#92702)

* 921872d97 Roll Skia from 762a01fd999e to 721388ecdbbe (4 revisions) (flutter/engine#29390)

* 5a6fa26e1 Make an interface for FlBinaryMessenger

* 3510312c6 Make an interface for FlPluginRegistrar

* a99f373c6 Make an interface for FlTextureRegistrar

* ed44967b8 Fix widget reference handling

* 98333c8bd Move mock objects into separate modules

* 71c8a9cd2 Can't test FlView on a server - no display

* 367e37165 Update licenses

* eb7401c7f Roll Skia from 721388ecdbbe to b5450fb9015b (2 revisions) (flutter/engine#29392)

* 111b80ef6 Inform the Dart VM when snapshots are safe to use with madvise(DONTNEED). (flutter/engine#29320)

* 684323cd3 [fuchsia] Remove unused sdk_ext.gni. (flutter/engine#29393)

* 3a0a63c2d Roll Skia from b5450fb9015b to 9fc189f1cbdf (2 revisions) (flutter/engine#29394)

wanglikun7342 committed:
feat: migrate macos/macos_device.dart to null safety (#92633)

gspencergoog committed:
Differentiate between TalkBack versions for semantics tests. (#92602)

Makes the android_semantics_testing test not fail for newer versions of Talkback.

Also, since the test now depends on pub_semver, upgraded the pubspec files.

The only substantive changes in the PR are in dev/integration_tests/android_semantics_testing/test_driver/main_test.dart

christopherfujino commented on packages/flutter_tools/lib/src/commands/analyze_base.dart:
```diff
@@ -143,19 +149,22 @@ class PackageDependency {
     assert(hasConflict);
     final List<String> targets = values.keys.toList();
-    targets.sort((String a, String b) => values[b].length.compareTo(values[a].length));
+    targets.sort((String a, String b) => values[b]?.length ?? 0 .compareTo(values[a]?.length ?? 0));
```
I think these should be `!.length` and not `?.length ?? 0`

WizzXu replied:
Thank you for your review. After analysis, `valuse [] `is not `null`, I have already modified the code in accordance with the recommendation.

christopherfujino commented on packages/flutter_tools/lib/src/commands/analyze_base.dart:
```diff
@@ -143,19 +149,22 @@ class PackageDependency {
-        if (source == canonicalSource) {
-          canonical = true;
+      final List<String>? targetList = values[target];
+      if (targetList != null) {
```
this should never be `null`, make `targetList` non-nullable instead.

jmagman replied:
(or use the `SplayMap` as suggested in https://github.com/flutter/flutter/pull/92542#discussion_r737736649)

christopherfujino replied:
:+1:

WizzXu replied:
Yes ,  I have modified the code.
```
      final List<String> targetList = values[target]!;
      final int count = targetList.length;
```

christopherfujino reviewed (commented):
two minor nits about using `!` rather than handling `null` cases that shouldn't be possible. this is looking pretty good though!

christopherfujino committed:
Fix analysis throwing string (#91435)

jmagman committed:
Migrate mdns_discovery and ios simulator to null safety (#92124)

jmagman committed:
Migrate android_device to null safety (#92128)

darrenaustin committed:
Removed primaryVariant from usage by the SnackBar. (#92443)

Commit:
[flutter_tools] xcresult parser. (#92604)

guidezpl committed:
Revert "Refactor ThemeData  (#91497)" (#92820)

This reverts commit 78689d347c40e7c0faebc1c5bcfac80e96262a65.

Commit:
Update lockfiles (#92800)

dheerajv09 committed:
Update Data Table Class with Borders Like Table Widgets (#36837) (#92670)

jmagman reviewed (commented):
This looks good once you get the test exemption from @Hixie

Commit:
Roll Gallery with updated Gradle lockfiles (#92850)

engine-flutter-autoroll committed:
Roll Engine from e781ed8b2980 to 7cee46cca464 (6 revisions) (#92847)

* 9f3ca93d8 Roll Dart SDK from 3b11f88c96a5 to 976f160b547f (3 revisions) (flutter/engine#29435)

* 1eec8c1e5 Roll Dart SDK from 3b11f88c96a5 to 976f160b547f (3 revisions) (flutter/engine#29436)

* 9c5f5d233 Roll Skia from 88b36ad61e80 to 2d76d7760f80 (3 revisions) (flutter/engine#29437)

* 9c88e86f5 Roll Skia from 2d76d7760f80 to 259ad7844587 (1 revision) (flutter/engine#29438)

* 27a7d2dd9 Roll Skia from 259ad7844587 to 39eccabdc3e4 (1 revision) (flutter/engine#29439)

* 7cee46cca Revert "Roll Dart SDK from 3b11f88c96a5 to 976f160b547f (3 revisions)" (flutter/engine#29440)

goderbauer committed:
Blinking cursor respects TickerMode (#92614)

jmagman committed:
Migrate doctor to null safety (#92131)

tgucio committed:
Update clipboard status on cut (#92167)

Cut now explicitly updates the clipboard status.

engine-flutter-autoroll committed:
Roll Engine from 7cee46cca464 to 43561d8820e0 (3 revisions) (#92855)

guidezpl committed:
Reland "Refactor ThemeData  (#91497)" (part 1) (#92822)

engine-flutter-autoroll committed:
Roll Engine from 43561d8820e0 to 8317c5eedb68 (2 revisions) (#92857)

GaryQian committed:
Migrate test to moved FlutterPlayStoreSplitApplication (#92593)

engine-flutter-autoroll committed:
Roll Engine from 8317c5eedb68 to 98e3216ab470 (1 revision) (#92860)

engine-flutter-autoroll committed:
Roll Engine from 98e3216ab470 to 2ea889c9af5c (1 revision) (#92866)

wanglikun7342 committed:
feat: mirgate test_data/compile_error_project.dart to null safety (#92808)

jmagman committed:
Remove globals_null_migrated.dart, move into globals.dart (#92861)

engine-flutter-autoroll committed:
Roll Engine from 2ea889c9af5c to 7797c76bb99c (1 revision) (#92868)

engine-flutter-autoroll committed:
Roll Engine from 7797c76bb99c to a4e700eb0c8b (2 revisions) (#92875)

engine-flutter-autoroll committed:
Roll Engine from a4e700eb0c8b to c674d2cbc6e0 (4 revisions) (#92886)

engine-flutter-autoroll committed:
Roll Engine from c674d2cbc6e0 to 76b4caeafe75 (1 revision) (#92892)

engine-flutter-autoroll committed:
Roll Engine from 76b4caeafe75 to 6095a1367846 (3 revisions) (#92904)

jensjoha committed:
[flutter_tools] Instruct compiler before processing bundle (#92255)

engine-flutter-autoroll committed:
Roll Engine from 6095a1367846 to 4dbdc0844425 (1 revision) (#92918)

keyonghan committed:
Marks Mac_ios platform_view_ios__start_up to be unflaky (#92916)

iskakaushik committed:
Revert "Roll Engine from 6095a1367846 to 4dbdc0844425 (1 revision) (#… (#92926)

mdebbar committed:
[web] Allow the usage of url strategies without conditional imports (#77103)

jmagman committed:
Migrate flutter_command to null safety (#92871)

godofredoc committed:
Update flutter localizations. (#92923)

jmagman committed:
Skip codesigning during native macOS integration tests (#92938)

engine-flutter-autoroll committed:
Roll Plugins from e51cc1df7b75 to 53fff22979d2 (7 revisions) (#92954)

jmagman committed:
Add Help menu to macOS create template (#92946)

WizzXu commented:
@Hixie This null safe mirgate pr already passed through CR，could I get a test exemption?

engine-flutter-autoroll committed:
Roll Engine from 6095a1367846 to f08694a57857 (15 revisions) (#92977)

engine-flutter-autoroll committed:
Roll Engine from f08694a57857 to 6ae304aeef1b (1 revision) (#92980)

engine-flutter-autoroll committed:
Roll Engine from 6ae304aeef1b to 535f1c4c1c49 (2 revisions) (#92984)

LongCatIsLooong committed:
Move text editing `Action`s to `EditableTextState` (#90684)

scheglov committed:
Fix a few prefer_const_declarations and avoid_redundant_argument_values. (#92929)

engine-flutter-autoroll committed:
Roll Engine from 535f1c4c1c49 to e9d92561b9a7 (2 revisions) (#92996)

jmagman committed:
Remove globals_null_migrated.dart, move into globals.dart (#92948)

engine-flutter-autoroll committed:
Roll Engine from e9d92561b9a7 to caf6d2997ddd (1 revision) (#93004)

engine-flutter-autoroll committed:
Roll Plugins from 53fff22979d2 to 2e102b8a8c41 (1 revision) (#93005)

markusaksli-nc committed:
Add `expandedScale` to `FlexibleSpaceBar` (#92160)

keyonghan committed:
Skip overall_experience_test.dart: flutter run writes and clears pidfile appropriately (#93026)

fluttergithubbot committed:
Marks Linux_android flutter_gallery__start_up to be unflaky (#93022)

engine-flutter-autoroll committed:
Roll Engine from caf6d2997ddd to fd0f83359dd8 (2 revisions) (#93028)

TahaTesser committed:
[Cupertino] Exclude border for last action item in `CupertinoContextMenu` (#92481)

jmagman committed:
Migrate channel, clean and a few other commands to null safety (#92950)

jmagman committed:
Migrate doctor, format, and a few other commands to null safety (#92952)

TahaTesser committed:
[ReorderableListView] Update doc with example to make it clear `proxyDecorator` can overriden to customise an item when it is being dragged. (#91837)

engine-flutter-autoroll committed:
Roll Engine from fd0f83359dd8 to 4f5e2968b639 (1 revision) (#93033)

fzyzcjy committed:
Fix typo that is introduced in hotfix 2.5.x (#92745)

yjbanov committed:
[web:tools] always use CanvasKit from the cache when building web apps (#93002)

jason-simmons committed:
Replace text directionality control characters with escape sequences in the semantics_tester (#93034)

markusaksli-nc committed:
Add `animationDuration` property to TabController (#91987)

HansMuller committed:
Added widgets/AppModel (#92297)

engine-flutter-autoroll committed:
Roll Engine from 4f5e2968b639 to c70041c85134 (4 revisions) (#93042)

engine-flutter-autoroll committed:
Roll Engine from c70041c85134 to bbc4615ecd63 (2 revisions) (#93043)

engine-flutter-autoroll committed:
Roll Engine from bbc4615ecd63 to e28b5df18c97 (1 revision) (#93046)

engine-flutter-autoroll committed:
Roll Engine from e28b5df18c97 to 32fff2f6e12b (1 revision) (#93051)

engine-flutter-autoroll committed:
Roll Plugins from 2e102b8a8c41 to e39a584ab08e (1 revision) (#93060)

jmagman committed:
Suppress created file list for new "flutter create" projects (#93041)

engine-flutter-autoroll committed:
Roll Engine from 32fff2f6e12b to 5140e30cec13 (1 revision) (#93061)

caseycrogers committed:
Allow Programmatic Control of Draggable Sheet (#92440)

engine-flutter-autoroll committed:
Roll Plugins from e39a584ab08e to d6ca8a368ce3 (1 revision) (#93066)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_7_last to be unflaky (#93020)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_0 to be unflaky (#93013)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_3 to be unflaky (#93016)

jmagman committed:
Update docs to point to assets-for-api-docs cupertino css file  (#92932)

andreidiaconu committed:
Reland engine display features (#89511)

zanderso committed:
Revert "Reland engine display features (#89511)" (#93098)

This reverts commit 058dfd49a0be0b13937a7c0d1c926dc6d60d47bf.

Commit:
Pin to specific plugin version in multidex test (#93148)

JBBx2016 committed:
Fixes issue where navigating to new route breaks FocusNode of previou… (#92615)

jakemac53 committed:
Update packages (#92924)

markusaksli-nc committed:
Add golden tests for default and overriden `expandedTitleScale` in `FlexibleSpaceBar` (#93059)

engine-flutter-autoroll committed:
Roll Plugins from d6ca8a368ce3 to 27eda487859c (3 revisions) (#93157)

andreidiaconu committed:
Reland: display features (#93117)

jmagman committed:
Migrate tracing to null safety (#92849)

jmagman committed:
Migrate custom_device to null safety (#92955)

jmagman committed:
Update minimum required version to Xcode 12.3 (#93094)

HansMuller committed:
Revert "Added widgets/AppModel" (#93161)

keyonghan committed:
Stop sending metrics to builder name (#92989)

Commit:
Bump Gallery version (#93173)

zanderso committed:
Revert "Reland: display features (#93117)" (#93204)

This reverts commit 9038fac96055aa0536fddf6deb31b823e361271e.

zanderso committed:
Revert "Bump Gallery version (#93173)" (#93207)

This reverts commit 36a0bc5b964cb2f49988bfd0410ef8601955f554.

darrenaustin committed:
Added `onDismiss` callback to ModalBarrier. (#83860)

TahaTesser committed:
[Cupertino] fix dark mode for `ContextMenuAction` (#92480)

jason-simmons committed:
Add the exported attribute to the Flutter Gallery manifest (#92733)

jmagman committed:
Migrate windows_device to null safety (#92956)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_1 to be unflaky (#93014)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_2 to be unflaky (#93015)

Commit:
[ci.yaml] Mark roller flaky (#93258)

xu-baolin committed:
Fix a scrollbar crash bug (#92657)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_6 to be unflaky (#93019)

tgucio committed:
Remove comment on caching line metrics (#93128)

engine-flutter-autoroll committed:
Roll Plugins from 27eda487859c to 5bf7fb0360aa (1 revision) (#93165)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_4 to be unflaky (#93017)

fluttergithubbot committed:
Marks Linux web_canvaskit_tests_5 to be unflaky (#93018)

engine-flutter-autoroll committed:
Roll Engine from 5140e30cec13 to 469d6f1a09f4 (58 revisions) (#93223)

* 422b15a80 Roll Skia from d9d9e21b311c to a8d38078a4f3 (1 revision) (flutter/engine#29521)

* afe1c980a Roll Fuchsia Mac SDK from SfdmlzXiU... to nkHhPcy3q... (flutter/engine#29522)

* b7f465178 Roll Fuchsia Linux SDK from 5-RB9TzQH... to m90mMA37b... (flutter/engine#29523)

* 15d5a23cf [Web] Fix BMP encoder (flutter/engine#29448)

* 3f55aece6 Fix FlutterPresentInfo struct_size doc string (flutter/engine#29524)

* b991af92f Roll Skia from a8d38078a4f3 to 7368c6d00b7c (5 revisions) (flutter/engine#29525)

* 93666e211 Fix isolate_configuration typo (flutter/engine#29318)

* 91033b9da Roll Skia from 7368c6d00b7c to a5030e9090e8 (3 revisions) (flutter/engine#29527)

* c2090c230 [iOS] Make sure spawnee's isGpuDisabled is set correctly when FlutterEngine spawn (flutter/engine#29361)

* e071f1d96 Roll Skia from a5030e9090e8 to 37afdbc22e89 (4 revisions) (flutter/engine#29528)

* ca1710a62 Roll Skia from 37afdbc22e89 to a05d3029ac65 (4 revisions) (flutter/engine#29532)

* 3364409b9 Reland Display Features support (Foldable and Cutout) (flutter/engine#29447)

* 4684d2327 [fuchsia] Add more logging for error cases. (flutter/engine#29397)

* 5cf7c390f Remove D3D9 fallback path (flutter/engine#29533)

* 98d31de63 [raster_cache] Increment access_count on Touch (flutter/engine#29529)

* cc6fc4a3f Re-enable A11yTreeIsConsistent with higher timeout (flutter/engine#29367)

* fb209165e Roll Skia from a05d3029ac65 to 37da672b14b7 (1 revision) (flutter/engine#29534)

* d229ec8d4 use SkMatrix.invert() instead of MatrixDecomposition to validate cache matrices (flutter/engine#29530)

* a92aebb09 Roll Clang Linux from 5N9a1nYj5... to UtjvZhwws... (flutter/engine#29535)

* 47dab8af3 Roll Dart SDK from 3b11f88c96a5 to f38618d5d0c0 (7 revisions) (flutter/engine#29537)

* 83ccad723 FragmentProgram constructed asynchronously (flutter/engine#29513)

* 2078baf24 Roll Fuchsia Linux SDK from m90mMA37b... to Ci-Vji1rx... (flutter/engine#29546)

* 6bf80930b Roll Fuchsia Mac SDK from nkHhPcy3q... to emi7COLIo... (flutter/engine#29547)

* 290d1792c Roll Skia from 37da672b14b7 to cf8daf79e710 (9 revisions) (flutter/engine#29548)

* 197ca594c [iOS] Fixes press key message leaks (flutter/engine#29354)

* 986b8d8ff Roll Skia from cf8daf79e710 to ae67f07a58a2 (1 revision) (flutter/engine#29549)

* 11010f002 Roll Skia from ae67f07a58a2 to 17616469ddf8 (1 revision) (flutter/engine#29551)

* dadc7b217 Roll Skia from 17616469ddf8 to 4322c7fec7e4 (3 revisions) (flutter/engine#29552)

* ed66091d9 don't build flutter SDK artifacts for armv7 (flutter/engine#28016)

* 484cc5c0e Roll Dart SDK from f38618d5d0c0 to 05febe0a7860 (5 revisions) (flutter/engine#29539)

* 387563aec Roll Skia from 4322c7fec7e4 to 1800d410df16 (1 revision) (flutter/engine#29553)

* 05ceba2e9 Roll Skia from 1800d410df16 to 725705f6630b (1 revision) (flutter/engine#29555)

* 2b142ef2c Roll Dart SDK from 05febe0a7860 to 38e7078fa2b7 (5 revisions) (flutter/engine#29556)

* 29da91f1f Roll Fuchsia Linux SDK from Ci-Vji1rx... to kHXT3xnTG... (flutter/engine#29557)

* da02a4b65 Roll Fuchsia Mac SDK from emi7COLIo... to 6BYh8qaYo... (flutter/engine#29558)

* 71fb57595 Roll Skia from 725705f6630b to deb9386be146 (3 revisions) (flutter/engine#29559)

* c5f572ac7 [Linux keyboard] Fix logical keys of up events are not regularized (flutter/engine#29550)

* 988ea435e Roll Dart SDK from 38e7078fa2b7 to d464cd3f2dc8 (5 revisions) (flutter/engine#29561)

* fdee74d86 Trace calls to Canvas::saveLayer (flutter/engine#29444)

* 953f718ee ios test script checks for `ios_test_flutter` artifacts (flutter/engine#29554)

* 145d89057 Roll Skia from deb9386be146 to 37bef2d300e4 (6 revisions) (flutter/engine#29563)

* cc9b174c2 Fix race condition introduced by background platform channels (flutter/engine#29377)

* 79b5e3c04 Roll Dart SDK from d464cd3f2dc8 to 996ef242a2c9 (1 revision) (flutter/engine#29564)

* 3eb6e151e fuchsia: Enable integration tests (flutter/engine#29565)

* ffa8b2582 Roll Skia from 37bef2d300e4 to 2417872a9993 (1 revision) (flutter/engine#29567)

* 9af2de970 Roll Dart SDK from 996ef242a2c9 to 5ccf755b37a4 (1 revision) (flutter/engine#29568)

* 16d6d1884 Roll Fuchsia Linux SDK from kHXT3xnTG... to uP2kJIngK... (flutter/engine#29569)

* 321bc5660 Roll Fuchsia Mac SDK from 6BYh8qaYo... to W9UXc2Fwx... (flutter/engine#29570)

* f0a8d4eda Roll Dart SDK from 5ccf755b37a4 to f6a43e5eb71d (1 revision) (flutter/engine#29571)

* 0be030362 Hide a11y elements when resigning active (flutter/engine#29566)

* 9a3195a84 Roll Fuchsia Linux SDK from uP2kJIngK... to aD3d4Kqmy... (flutter/engine#29572)

* b3cbee58c Roll Fuchsia Mac SDK from W9UXc2Fwx... to rIpW1050J... (flutter/engine#29573)

* df686d263 Revert "Reland Display Features support (Foldable and Cutout) (#29447)" (flutter/engine#29574)

* e81273129 Revert dart to 38e7078fa2b7 (flutter/engine#29575)

* 049074ee9 Roll Skia from 2417872a9993 to cd7220e7686c (2 revisions) (flutter/engine#29577)

* e6a51a411 Roll Fuchsia Mac SDK from rIpW1050J... to TOmxgL3av... (flutter/engine#29578)

* 8f272d98c Roll Fuchsia Linux SDK from aD3d4Kqmy... to ZniYyCw7U... (flutter/engine#29579)

* 469d6f1a0 Revert "[Web] Fix BMP encoder (#29448)" (flutter/engine#29580)

fluttergithubbot committed:
Marks Linux_android flutter_gallery__start_up_delayed to be unflaky (#93251)

fluttergithubbot committed:
Marks Mac_android run_release_test to be unflaky (#93252)

fluttergithubbot committed:
Marks Mac_ios integration_ui_ios_textfield to be unflaky (#93254)

yjbanov committed:
[web] exclude bitrotten context_menu_action_test.dart from CanvasKit shard (#93268)

keyonghan committed:
mark flaky (#93266)

xu-baolin committed:
Do not crash when dragging ReorderableListView  with two fingers simultaneously (#88508)

guidezpl committed:
Reland "Refactor ThemeData (#91497)" (part 2)  (#92970)

christopherfujino committed:
[flutter_conductor] Rewrite writeStateToFile to be in an overidable method. (#93171)

HansMuller committed:
Added SharedAppData to the widgets library (#93175)

yjbanov committed:
Revert "mark flaky (#93266)" (#93273)

darrenaustin committed:
Force the color used by WidgetApp's Title to be fully opaque. (#93267)

HansMuller committed:
Add Border customization to CheckboxListTile (reprise) (#93271)

jpnurmi committed:
[gen_l10n] retain full output file suffix (#88362)

Yugue committed:
added cleanContext (#93264)

* added cleanContext

* using async

* added one more test

GaryQian committed:
Dynamic logcat piping for deferred components test (#93080)

jmagman committed:
In CupertinoIcons doc adopt diagram and fix broken glyph map link (#93087)

yjbanov committed:
[web] add image decoder benchmark (#93174)

christopherfujino committed:
[flutter_conductor] ensure release branch point is always tagged (#93082)

engine-flutter-autoroll committed:
Roll Engine from 469d6f1a09f4 to 1ae721c0230e (11 revisions) (#93284)

fluttergithubbot committed:
Marks Linux_android flutter_gallery__image_cache_memory to be unflaky (#93021)

engine-flutter-autoroll committed:
Roll Engine from 1ae721c0230e to 0b3ac9431351 (1 revision) (#93294)

engine-flutter-autoroll committed:
Roll Engine from 0b3ac9431351 to 5fcc59bbc290 (2 revisions) (#93296)

engine-flutter-autoroll committed:
Roll Engine from 5fcc59bbc290 to 162d8508f874 (2 revisions) (#93299)

gspencergoog committed:
Reland: "Update outdated runners in the benchmarks folder (#91126)" (#92535)

GaryQian committed:
Use adb variable instead of direct command (#93293)

zanderso committed:
Revert "Reland: "Update outdated runners in the benchmarks folder (#91126)" (#92535)" (#93323)

This reverts commit bfe9c59831ebf60d697b088a2da35897a466397a.

engine-flutter-autoroll committed:
Roll Engine from 162d8508f874 to 766f3a5bec7b (2 revisions) (#93306)

engine-flutter-autoroll committed:
Roll Plugins from 5bf7fb0360aa to 638dd07f51fa (1 revision) (#93309)

GaryQian committed:
Track timeout from app run start in deferred components integration test (#93307)

creativecreatorormaybenot committed:
Improve ProgressIndicator value range docs (#88800)

goderbauer committed:
Do not rebuild when TickerMode changes (#93166)

camsim99 committed:
Add warning for bumping Android SDK version to match plugins (#92451)

engine-flutter-autoroll committed:
Roll Plugins from 638dd07f51fa to 68d222342e39 (2 revisions) (#93341)

engine-flutter-autoroll committed:
Roll Engine from 766f3a5bec7b to 76bf5a1ad029 (5 revisions) (#93327)

GaryQian committed:
Exit on deprecated v1 embedding when trying to run or build (#92901)

darrenaustin committed:
Replaced the reference to `primaryVariant` in code example as it will soon be deprecated. (#93351)

engine-flutter-autoroll committed:
Roll Plugins from 68d222342e39 to 1a2fbe878d86 (1 revision) (#93354)

zanderso committed:
Revert "Exit on deprecated v1 embedding when trying to run or build (#92901)" (#93365)

This reverts commit 6153bbef3e8e8202a0f32650406c879a40775175.

natebosch committed:
Ignore upcoming warning for unnecessary override (#93353)

evan361425 committed:
Use num on plural (#93228)

godofredoc committed:
Pin visual studio version used in the framework. (#93356)

engine-flutter-autoroll committed:
Roll Engine from 76bf5a1ad029 to 3af330d5c098 (27 revisions) (#93414)

engine-flutter-autoroll committed:
Roll Plugins from 1a2fbe878d86 to d9d87e9f8809 (1 revision) (#93419)

engine-flutter-autoroll committed:
Roll Engine from 3af330d5c098 to 5c1b2d9af0fd (4 revisions) (#93421)

engine-flutter-autoroll committed:
Roll Plugins from d9d87e9f8809 to 0cfbe1779ecb (3 revisions) (#93424)

TahaTesser committed:
[CupertinoActivityIndicator] Add `color` parameter (#92172)

engine-flutter-autoroll committed:
Roll Engine from 5c1b2d9af0fd to e5f05693010f (3 revisions) (#93428)

simolus3 committed:
Improve error message for uninitialized channel (#92758)

gspencergoog committed:
Adds tool warning log level and command line options to fail on warning/error output (#92031)

tgucio committed:
Use baseline value to get position in next line (#93129)

engine-flutter-autoroll committed:
Roll Plugins from 0cfbe1779ecb to 112fe4eb497a (2 revisions) (#93435)

engine-flutter-autoroll committed:
Roll Engine from e5f05693010f to 6abca2895599 (1 revision) (#93436)

stuartmorgan-g committed:
Add support for Visual Studio 2022 (#93426)

engine-flutter-autoroll committed:
Roll Engine from 6abca2895599 to d5cadd28b0bf (3 revisions) (#93444)

nt4f04uNd committed:
RangeMaintainingScrollPhysics remove overscroll maintaining when grow (#90608)

engine-flutter-autoroll committed:
Roll Plugins from 112fe4eb497a to d09abd526375 (2 revisions) (#93447)

engine-flutter-autoroll committed:
Roll Plugins from d09abd526375 to d22be768e9d5 (1 revision) (#93458)

engine-flutter-autoroll committed:
Roll Plugins from d22be768e9d5 to a103bcd3327e (1 revision) (#93484)

HelloCore committed:
feat: enable flavor option on test command (#89045)

engine-flutter-autoroll committed:
Roll Engine from d5cadd28b0bf to 2a76e06247df (9 revisions) (#93487)

keyonghan committed:
Marks Mac_android run_release_test to be flaky (#93490)

engine-flutter-autoroll committed:
Roll Engine from 2a76e06247df to 26b3c41ad795 (2 revisions) (#93496)

Swiftaxe committed:
Removed no-shuffle tag and fixed leak in flutter_goldens_test.dart (#92374)

engine-flutter-autoroll committed:
Roll Engine from 26b3c41ad795 to aaacb670af16 (1 revision) (#93503)

kenzieschmoll committed:
Add DevTools version to `flutter --version` and `flutter doctor -v` output. (#93065)

* Add DevTools version to `flutter --version` output.

* review comments

* fix test

* add explanatory comment

* dummy commit to kick bots

Co-authored-by: Christopher Fujino <christopherfujino@gmail.com>

smitsk committed:
Fix text cursor or input can be outside of the text field (#91698)

GaryQian committed:
Reland "Exit on deprecated v1 embedding when trying to run or build (#92901)" (#93386)

goderbauer committed:
Use navigator instead of overlay as TickerProvider for ModalBottomSheet (#93499)

yjbanov committed:
Revert "Reland "Exit on deprecated v1 embedding when trying to run or build (#92901)" (#93386)" (#93518)

This reverts commit eabb7cbc341e51fc138dc02e974eb0dc35178598.

It broke multiple tests.

engine-flutter-autoroll committed:
Roll Engine from aaacb670af16 to e022c9283c40 (4 revisions) (#93510)

jason-simmons committed:
Restore Cache.flutterRoot after overriding it in a test (#93555)

Fixes https://github.com/flutter/flutter/issues/93553

00SunnyDay00 committed:
add --extra-gen-snapshot-options support for build_aar command. (#91590)

gaaclarke committed:
Fatten up multiple_flutters memory footprint (#93350)

Increased the memory size of multiple_flutters to help identify regressions in performance

This is not a regression and will affect benchmark's readings.

creativecreatorormaybenot committed:
Fix typo in AnimatedWidgetBaseState (#93536)

engine-flutter-autoroll committed:
Roll Plugins from a103bcd3327e to ace9e78bd3c3 (2 revisions) (#93560)

jmagman committed:
Ad-hoc codesign Flutter.framework when code signing is disabled (#93556)

engine-flutter-autoroll committed:
Roll Engine from e022c9283c40 to 701d66d109e2 (16 revisions) (#93563)

engine-flutter-autoroll committed:
Roll Engine from 701d66d109e2 to 85e06f206389 (3 revisions) (#93568)

christopherfujino committed:
[flutter_tools] Catch lack of flutter tools source missing (#93168)

engine-flutter-autoroll committed:
Roll Engine from 85e06f206389 to c19137fd56fb (1 revision) (#93572)

GaryQian committed:
Reland "Exit on deprecated v1 embedding when trying to run or build" (#92901) (#93566)

engine-flutter-autoroll committed:
Roll Engine from c19137fd56fb to ea8386ff6547 (2 revisions) (#93574)

engine-flutter-autoroll committed:
Roll Engine from ea8386ff6547 to 140bd9399e63 (1 revision) (#93579)

christopherfujino committed:
[flutter_conductor] allow --force to override validations on start sub-command (#93514)

engine-flutter-autoroll committed:
Roll Plugins from ace9e78bd3c3 to 8a95012fb037 (1 revision) (#93587)

nayeemtby committed:
Add scaleX and scaleY Parameters in Transform.Scale. (#92343)

engine-flutter-autoroll committed:
Roll Plugins from 8a95012fb037 to 31268aabbaa9 (1 revision) (#93628)

Piinks committed:
Reland Remove BottomNavigationBarItem.title deprecation (#92197)

Piinks committed:
Close tree if unapproved golden images get in (#93516)

engine-flutter-autoroll committed:
Roll Engine from 140bd9399e63 to 2962099077b3 (26 revisions) (#93675)

jason-simmons committed:
Handle empty text with no line metrics in RenderEditable._lineNumberFor (#93578)

Piinks committed:
Skip negative web image test (#93722)

jmagman commented:
> @Hixie This null safe mirgate pr already passed through CR，could I get a test exemption?

You need to ask for an exemption in Discord.

> If you need an exemption to this rule, contact Hixie on the #hackers channel in Chat.

engine-flutter-autoroll committed:
Roll Plugins from 31268aabbaa9 to 1e85f320695d (9 revisions) (#93694)

Piinks committed:
Skip the right web golden file test (#93740)

werainkhatri committed:
Updated IconButton.iconSize to get value from theme (#87643)

Reekoooo committed:
Remove shade50 getter from MaterialAccentColor (#93507)

justinmc committed:
Desktop edge scrolling (#93170)

HansMuller committed:
ChipThemeData is now conventional (#93452)

guidezpl committed:
Deprecate `primaryColorBrightness` (#93396)

TahaTesser committed:
Add `CupertinoButton` interactive example (#93625)

hterkelsen committed:
Fix analysis error in icon button test (#93749)

Letiste committed:
fix: inline templatized see also (#93600)

fluttergithubbot committed:
Marks Mac native_ui_tests_macos to be unflaky (#93664)

gspencergoog committed:
Fix CallbackShortcuts to only call shortcuts when triggered (#93693)

vlkonoshenko committed:
fix: removed unnecessary checks (#93735)

engine-flutter-autoroll committed:
Roll Plugins from 1e85f320695d to e6ff3527c3c1 (1 revision) (#93761)

engine-flutter-autoroll committed:
Roll Engine from 2962099077b3 to ace1f5f2997b (13 revisions) (#93769)

engine-flutter-autoroll committed:
Roll Engine from ace1f5f2997b to 47828a664fd7 (2 revisions) (#93775)

engine-flutter-autoroll committed:
Roll Plugins from e6ff3527c3c1 to 4ccc90493235 (2 revisions) (#93788)

engine-flutter-autoroll committed:
Roll Engine from 47828a664fd7 to 88644c8dcb45 (1 revision) (#93789)

engine-flutter-autoroll committed:
Roll Engine from 88644c8dcb45 to 6e2aafeb4b50 (2 revisions) (#93796)

Merrit committed:
Fix typo in Shortcuts doc comment (#93483)

TahaTesser committed:
allow full locale  in .arb files (#93401)

HansMuller committed:
Revert "allow full locale  in .arb files (#93401)" (#93806)

This reverts commit 16f441188c0ba387da3ea1141ae6fae47306167e.

DanTup committed:
Ensure the flutter_gen project is correctly created as part of "flutter create" (#92975)

engine-flutter-autoroll committed:
Roll Plugins from 4ccc90493235 to 8f55fb68d0e9 (1 revision) (#93802)

engine-flutter-autoroll committed:
Roll Engine from 6e2aafeb4b50 to d389ae475cde (2 revisions) (#93808)

engine-flutter-autoroll committed:
Roll Engine from d389ae475cde to 29ea28d67b2d (1 revision) (#93815)

Swiftaxe committed:
Fixed leak and removed no-shuffle tag in window_test.dart (#93604)

fluttergithubbot committed:
Marks Mac_android android_semantics_integration_test to be flaky (#93255)

engine-flutter-autoroll committed:
Roll Engine from 29ea28d67b2d to c30375b830ef (1 revision) (#93820)

jmagman committed:
Migrate build_system targets to null safety (#92869)

engine-flutter-autoroll committed:
Roll Plugins from 8f55fb68d0e9 to 88c69ed86147 (1 revision) (#93832)

engine-flutter-autoroll committed:
Roll Plugins from 88c69ed86147 to 349a858048db (1 revision) (#93836)

WizzXu commented:
> > @Hixie This null safe mirgate pr already passed through CR，could I get a test exemption?
> 
> You need to ask for an exemption in Discord.
> 
> > If you need an exemption to this rule, contact Hixie on the #hackers channel in Chat.

Sorry，i  I did not find `#hackers` in Discord，I have sent an email to Hixie, I don't know if I can get a reply.

engine-flutter-autoroll committed:
Roll Engine from c30375b830ef to a3ef1d5351f4 (5 revisions) (#93839)

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_base.dart:
```diff
@@ -81,16 +79,23 @@ abstract class AnalyzeBase {
 
   bool get isFlutterRepo => argResults['flutter-repo'] as bool;
-  String get sdkPath => argResults['dart-sdk'] as String ?? artifacts.getHostArtifact(HostArtifact.engineDartSdkPath).path;
+  String get sdkPath {
+    final String? dartSdk = argResults['dart-sdk'] as String?;
+    if (dartSdk is String) {
+      return dartSdk;
+    } else {
+      return artifacts.getHostArtifact(HostArtifact.engineDartSdkPath).path;
+    }
+  }
```
```suggestion
  String get sdkPath {
    final String? dartSdk = argResults['dart-sdk'] as String?;
    if (dartSdk != null) {
      return dartSdk;
    }
    return artifacts.getHostArtifact(HostArtifact.engineDartSdkPath).path;
  }
```

or just:

```suggestion
String get sdkPath => argResults['dart-sdk'] as String? ?? artifacts.getHostArtifact(HostArtifact.engineDartSdkPath).path;
```

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_base.dart:
```diff
@@ -129,11 +134,12 @@ class PackageDependency {
       for (final String source in targetSources) {
         assert(globals.fs.path.isAbsolute(source));
-        if (globals.fs.path.isWithin(Cache.flutterRoot, source)) {
+        if (flutterRoot != null && globals.fs.path.isWithin(flutterRoot, source)) {
```
we assert that it's not null above, so presumably there's no need to check it again here

Hixie commented on packages/flutter_tools/lib/src/commands/analyze_base.dart:
```diff
@@ -129,11 +134,12 @@ class PackageDependency {
   bool get hasConflictAffectingFlutterRepo {
-    assert(globals.fs.path.isAbsolute(Cache.flutterRoot));
+    final String? flutterRoot = Cache.flutterRoot;
+    assert(flutterRoot != null && globals.fs.path.isAbsolute(flutterRoot));
```
You can get both the "assert not null" _and_ the local promotion using this style:

```suggestion
    flutterRoot!;
    assert(globals.fs.path.isAbsolute(flutterRoot));
```

Hixie commented:
test-exempt: code refactor with no semantic change

moko256 committed:
Win32 template: Use {{projectName}} for FileDescription instead of {{description}} (#91959)

Hixie commented:
Looks like some of the tests are failing, so maybe the refactor had more of a semantic change than was intended :-)

WizzXu commented:
> Looks like some of the tests are failing, so maybe the refactor had more of a semantic change than was intended :-)

According to the error log analysis, it seems to be a problem of test engineering compatibility, not the problem of this modified code.

Error Log
```
04:18 +24 -2: test/integration.shard/multidex_build_test.dart: simple build apk without FlutterMultiDexApplication fails [E]                                                                           
  Expected: contains 'Cannot fit requested classes in a single dex file'
    Actual: '/opt/s/w/ir/x/t/flutter_run_test.E3XXz5/android/app/src/main/AndroidManifest.xml Error:\n'
              '\tuses-sdk:minSdkVersion 16 cannot be smaller than version 19 declared in library [com.google.firebase:firebase-firestore:24.0.0] /opt/s/w/ir/cache/gradle/caches/transforms-2/files-2.1/eb1f62a445787c498ef3d7cdac9e7898/jetified-firebase-firestore-24.0.0/AndroidManifest.xml as the library might be using APIs not available in 16\n'
              '\tSuggestion: use a compatible library with a minSdk of at most 16,\n'
              '\t\tor increase this project\'s minSdk version to at least 19,\n'
              '\t\tor use tools:overrideLibrary="com.google.firebase.firestore" to force usage (may lead to runtime failures)\n'
              '\n'
              'FAILURE: Build failed with an exception.\n'
              '\n'
              '* What went wrong:\n'
              'Execution failed for task \':app:processDebugMainManifest\'.\n'
              '> Manifest merger failed : uses-sdk:minSdkVersion 16 cannot be smaller than version 19 declared in library [com.google.firebase:firebase-firestore:24.0.0] /opt/s/w/ir/cache/gradle/caches/transforms-2/files-2.1/eb1f62a445787c498ef3d7cdac9e7898/jetified-firebase-firestore-24.0.0/AndroidManifest.xml as the library might be using APIs not available in 16\n'
              '  \tSuggestion: use a compatible library with a minSdk of at most 16,\n'
              '  \t\tor increase this project\'s minSdk version to at least 19,\n'
              '  \t\tor use tools:overrideLibrary="com.google.firebase.firestore" to force usage (may lead to runtime failures)\n'
              '\n'
              '* Try:\n'
              'Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.\n'
              '\n'
              '* Get more help at https://help.gradle.org\n'
              '\n'
              'BUILD FAILED in 19s\n'
              'Gradle task assembleDebug failed with exit code 1\n'
              ''
  
  package:test_api                                      expect
  test/integration.shard/multidex_build_test.dart 57:5  main.<fn>
  

04:18 +24 -2: loading test/integration.shard/multidex_build_test.dart                                                                                                                                  
Consider enabling the flag chain-stack-traces to receive more detailed exceptions.
For example, 'dart test --chain-stack-traces'.
```

engine-flutter-autoroll committed:
Roll Plugins from 349a858048db to 58d215dd926c (2 revisions) (#93886)

jmagman commented:
Rebase your change onto the fix at https://github.com/flutter/flutter/pull/93148.

WizzXu committed:
Merge branch 'flutter:master' into master_null_safe

jmagman commented:
Because this was briefly merged from some other branch the Google CLA bot is confused.  You may be best off closing and recreating this PR.

WizzXu commented:
> recreating

OK, I have recreating this PR  #93954

Piinks commented:
Thanks for creating another PR @WizzXu! I am going to close this one then per your last comment. :)
